azurerm/internal/services/hdinsight/hdinsight_hbase_cluster_resource.go

```diff
@@ -139,7 +141,14 @@ func resourceArmHDInsightHBaseClusterCreate(d *schema.ResourceData, meta interfa
 	componentVersions := expandHDInsightHBaseComponentVersion(componentVersionsRaw)
 
 	gatewayRaw := d.Get("gateway").([]interface{})
-	gateway := azure.ExpandHDInsightsConfigurations(gatewayRaw)
+	configurations := azure.ExpandHDInsightsConfigurations(gatewayRaw)
+
+	if metastoresRaw, ok := d.GetOkExists("metastores"); ok {
+		metastores := expandHDInsightsMetastore(metastoresRaw.([]interface{}))
+		for k, v := range metastores {
+			configurations[k] = v
+		}
+	}
```

Comment:
`GetOkExists` is only intended for bools, and is also being deprecated in the plugin SDK in future.  We're working through removing / update other occurrences at the moment.  The following should be in line with the rest of the changes (along with the change to `expandHDInsightsMetastore` in 6809, so I'll make this PR dependent on that one)

```suggestion
	metastoresRaw, ok := d.Get("metastores").([]interface{})
	metastores := expandHDInsightsMetastore(metastoresRaw)
		for k, v := range metastores {
			configurations[k] = v
		}
	}
```